src/js/background/background.js

```diff
@@ -361,7 +361,11 @@ async function displayBrowserActionBadge() {
 browser.runtime.onMessage.addListener(async (m, sender, _sendResponse) => {
   let response;
   const currentPage = await getCurrentPage();
-  const url = new URL(currentPage.url);
+  let url = false;
```

Comment:
LMK if there's a better way to handle this error than using `let` and resetting inside an `if` statement.

Reply:
Better is subjective, but an _alternative_ way is

```suggestion
  const url = typeof currentPage.url === "string" ? new URL(currentPage.url) : false;
```

_Another_ alternative way is to remove this line altogether, and below do:

```diff
- if (url) { response = url.hostname }
+ if (url) { response = (new URL(currentPage.url))l.hostname }
```

But they're all fine by me :smile:

Reply:
Solid. I was able to do a mix of the two. Much cleaner! 

_(and comment added)_